docs/bir-spec/src/test/java/org/ballerinalang/birspec/BIRTestUtils.java

```diff
@@ -42,9 +53,44 @@
 
     private static final String TEST_RESOURCE_ROOT = "test-src/";
 
+    private static final String LANG_LIB_TEST_SRC_ROOT = "../../langlib/langlib-test/src/test/resources/test-src";
+
+    private static final Path TEST_ROOT_PATH = Paths.get(LANG_LIB_TEST_SRC_ROOT);
+
+    static void validateBIRSpec() throws IOException {
+
+        assert TEST_ROOT_PATH.toFile().exists();
+        List<String> testSources = Files.walk(TEST_ROOT_PATH)
+                .filter(file -> Files.isRegularFile(file))
+                .map(file -> file.toAbsolutePath().normalize().toString())
+                .filter(file -> file.endsWith(".bal") && !file.contains("negative") && !file.contains("subtype"))
+                .collect(Collectors.toList());
```

Comment:
Better to break this into two methods and use ` @DataProvider` to link them. that way you can run one file by file if there is a failed one.

Reply:
ack, will check on it.